src/InputWithLabel.js

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useRef } from "react";
+
+function InputWithLabel({
+  id,
+  name,
+  type = "text",
+  value,
+  children,
+  onInputChange,
+  placeholder,
+  autoFocus,
+}) {
+  const inputRef = useRef();
+  useEffect(() => {
+    if (autoFocus) {
+      inputRef.current.focus();
+    }
+  });
+
+  return (
+    <>
+      <label htmlFor={id}>{children}</label>
+      <input
+        ref={inputRef}
+        name={name}
+        id={id}
+        value={value}
+        placeholder={placeholder}
+        type={type}
+        onChange={onInputChange}
+      ></input>
```

Comment:
Nitpick:
Input elements will never have children, so they can be self closing tags:
```js
<input ref={inputRef} name={name} ...etc  />
```